api/src/main/java/org/pmiops/workbench/config/WebMvcConfig.java

```diff
@@ -48,18 +49,28 @@
   @Bean
   @RequestScope(proxyMode = ScopedProxyMode.DEFAULT)
   public UserAuthentication userAuthentication() {
-    return (UserAuthentication) SecurityContextHolder.getContext().getAuthentication();
+    Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+    if (auth == null) {
+      return null;
```

Comment:
I really think null beans are an anti-pattern. I expect them to be there when they're injected, but maybe i've just. been naive. If we do. need this, then I recommend moving to `OptionalProvider`, so it's. obvious.

Can we not make UserAuthentication. and Userinfoplus a bit smarter so they don't have to be. null?

Reply:
I agree. My favorite solution is a class like `RequestAuthenticationInfo` because

- The injectee itself is never null or empty
- It's obvious when I declare a `Provider<RequestAuthenticationInfo>` that more than one request context needs to be supported
- It can be adopted incrementally instead of as a Big Bang refactoring.

The point on failed binding is well taken, though I think on balance I much prefer that risk to allowing injected nulls.